src/analysis/sensitivity.jl

```diff
@@ -32,6 +32,52 @@ function ks_statistic(ks::ApproximateKSTest)::Float64
     return sqrt(n) * ks.δ
 end
 
+"""
+    _get_factor_spec(model_spec::DataFrame, factors::Vector{String})
+
+Get model spec for specified factors.
+
+# Arguments 
+- model_spec : Model specification, from `model_spec(domain)` or `rs.model_spec`
+- factors : Parameters considered for sensitivity analysis
+"""
+function _get_factor_spec(model_spec::DataFrame, factors::Vector{Symbol})
+    factors_to_assess = model_spec.fieldname .∈ [factors]
+    foi_attributes = Symbol[:fieldname, :ptype, :lower_bound, :upper_bound]
+    return model_spec[factors_to_assess, foi_attributes]
+end
+
+"""
+    _category_bins(S::Int64, foi_spec::DataFrame, foi_cat::BitVector)
+
+Get number of binnings for categorical variables.
+
+# Arguments
+- S : Number of bins
+- foi_spec : Model specification for factors of interest
+- foi_cat : Vector of length `foi_cat.fieldname` which contains true where the factor is categorical and false otherwise
+"""
+function _category_bins(S::Int64, foi_spec::DataFrame, foi_cat::BitVector)
+    max_bounds = maximum(foi_spec[foi_cat, :upper_bound] .- foi_spec[foi_cat, :lower_bound])
+    return round(Int64, max(S, max_bounds))
+end
+
+"""
+    _get_cat_quantile(foi_spec::DataFrame, fact_cat::String, steps::Vector{Float64})
+
+Get quantile for categorical variable, fact_cat.
+
+# Arguments
+- foi_spec : Model specification for factors of interest
+- foi_cat : Vector of length `foi_cat.fieldname` which contains true where the factor is categorical and false otherwise
+- steps : Number of steps for defining binnings
```

Comment:
Arguments need to be updated and backticks added

Reply:
Sorry, not sure why I missed this